lighthouse-core/gather/driver.js

```diff
@@ -496,6 +496,16 @@ class Driver {
     });
   }
 
+  /**
+   * Returns a promise that resolve when a frame has been navigated.
+   * Used for detecting that our about:blank reset has been completed.
+   */
+  _waitForFrameNavigated() {
+    return new Promise(resolve => {
```

Comment:
at some point we should add a driver method that wraps `once` with a promise. Seems like a lot of our uses of it end up like this